.github/workflows/pr.yml

```diff
@@ -0,0 +1,61 @@
+name: UI5-Language-Assistant
+
+on:
+  
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    env:
+      HUSKY_SKIP: "true"
+      # NODE_OPTIONS: "--openssl-legacy-provider"
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+        os: [macos-latest, ubuntu-latest, windows-latest]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'yarn'
+    - name: Setup yarn
+      run: |
+            npm install -g yarn
+            yarn
+    - name: Build Node v18 - Linux
+      if: ${{ (matrix.node-version == '18.x') && (runner.os == 'Linux') }}
+      run: xvfb-run -a yarn run ci
+      env:
+          NODE_OPTIONS: "--openssl-legacy-provider"
+    - name: Build Node v18 - Others OS
+      if: ${{ (matrix.node-version == '18.x') && (runner.os != 'Linux') }}
+      run: yarn run ci
+      env:
+          NODE_OPTIONS: "--openssl-legacy-provider"
+    - name: Build Node v14/16 - Linux
+      if: ${{ (matrix.node-version != '18.x') && (runner.os == 'Linux') }}
+      run: xvfb-run -a yarn run ci
+    - name: Build Node v14/16 - Other OS
+      if: ${{ (matrix.node-version != '18.x') && (runner.os != 'Linux') }}
+      run: yarn run ci
+  
+  compliance:
```

Comment:
does not this mean the `compliance` step would run for each "element" of the build matrix? This seems like a waste, I would recommand to only run it once in a separate github actions step.